### PR TITLE
[serve] [ray_client] Basic test for using Ray client with Serve

### PR DESCRIPTION
**Pull request by edoakes**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by edoakes** [comment]
@barakmich ran into a few strange issues here as you can see in the comments on the test. Unfortunately I wasn't able to make a simple repro without Serve.

File: python/ray/serve/tests/test_ray_client.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import random
+import subprocess
+
+import pytest
+
+import requests
```
Comment (simon-mo) [inline python/ray/serve/tests/test_ray_client.py:6]:
```suggestion
import pytest
import requests
```

**Review by simon-mo — APPROVED** [review]
😢 so many lines commented out

**Comment by edoakes** [comment]
@simon-mo yeah, not ideal. Want to make sure to get a test running ASAP to avoid regressions though.

**Comment by simon-mo** [comment]
travis failed with failure to pickle pydantic classes.

https://travis-ci.com/github/ray-project/ray/jobs/484691758#L1008

should we just revisit this PR when pydantic dependency is confined to just the user facing API?

**Review by simon-mo — APPROVED** [review]

**Pull request merged**